### PR TITLE
Implement Countable interface for all Listings

### PR DESCRIPTION
**Pull request by BlackbitNeueMedien**
This PR implements PHP's Countable interface for all Listings, so you can do something like this:
```php
$listing = new Listing();
$listing->addConditionParam('column = ?', 123);
if(count($listing) === 0) {
  echo "No entries found";
}
```
Internally the `getTotalCount()` method of the Listing DAO gets called. Most of them already had this method - where it was missing this PR adds it.

### CODE REVIEW

File: models/GridConfigShare/Listing/Dao.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ public function getTotalCount()
         $amount = 0;
 
         try {
-            $amount = (int) $this->db->fetchOne('SELECT COUNT(*) as amount FROM gridconfig_shares ' . $this->getCondition(), $this->model->getConditionVariables());
+            $amount = (int) $this->db->fetchOne('SELECT COUNT(*) FROM gridconfig_shares ' . $this->getCondition(), $this->model->getConditionVariables());
```
Comment (dvesh3) [inline models/GridConfigShare/Listing/Dao.php:56]: why not return directly from here?

File: models/Dependency.php [hunk 2]
```diff
@@ -118,7 +118,7 @@ public function getRequires($offset = null, $limit = null)
      * @param int|null $offset
      * @param int|null $limit
      *
-     * @return array
+     * @return array{type: string, id: int}
```
Comment (dvesh3) [inline models/Dependency.php:121]: related?

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Listing/Dao.php [hunk 3]
```diff
@@ -36,8 +36,10 @@ public function load()
 
     public function getTotalCount()
     {
-        $amount = $this->db->fetchRow('SELECT COUNT(*) as amount FROM `' . \Pimcore\Bundle\EcommerceFrameworkBundle\CartManager\CartCheckoutData\Dao::TABLE_NAME . '`' . $this->getCondition());
-
-        return $amount['amount'];
+        try {
+            return $this->db->fetchOne('SELECT COUNT(*) FROM `' . \Pimcore\Bundle\EcommerceFrameworkBundle\CartManager\CartCheckoutData\Dao::TABLE_NAME . '`' . $this->getCondition());
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Listing/Dao.php:40]: cast?

File: bundles/EcommerceFrameworkBundle/CartManager/CartItem/Listing/Dao.php [hunk 4]
```diff
@@ -40,9 +40,11 @@ public function load()
 
     public function getTotalCount()
     {
-        $amount = $this->db->fetchRow('SELECT COUNT(*) as amount FROM `' . \Pimcore\Bundle\EcommerceFrameworkBundle\CartManager\CartItem\Dao::TABLE_NAME . '`' . $this->getCondition());
-
-        return $amount['amount'];
+        try {
+            return $this->db->fetchOne('SELECT COUNT(*) FROM `' . \Pimcore\Bundle\EcommerceFrameworkBundle\CartManager\CartItem\Dao::TABLE_NAME . '`' . $this->getCondition());
```
Comment (dvesh3) [inline bundles/EcommerceFrameworkBundle/CartManager/CartItem/Listing/Dao.php:44]: cast?

File: lib/Model/Listing/AbstractListing.php [hunk 5]
```diff
@@ -571,4 +571,14 @@ public function rewind()
         $this->getData();
         reset($this->data);
     }
+
+    public function count()
```
Comment (dvesh3) [inline lib/Model/Listing/AbstractListing.php:575]: phpdoc

File: models/ImportConfigShare/Listing/Dao.php [hunk 6]
```diff
@@ -53,7 +53,7 @@ public function getTotalCount()
         $amount = 0;
 
         try {
-            $amount = (int) $this->db->fetchOne('SELECT COUNT(*) as amount FROM importconfig_shares ' . $this->getCondition(), $this->model->getConditionVariables());
+            $amount = (int) $this->db->fetchOne('SELECT COUNT(*) FROM importconfig_shares ' . $this->getCondition(), $this->model->getConditionVariables());
```
Comment (dvesh3) [inline models/ImportConfigShare/Listing/Dao.php:56]: direct?

File: models/Dependency.php (hunk 2 above)
Comment (BlackbitNeueMedien) [reply models/Dependency.php]: Nope. Reverted.

File: models/GridConfigShare/Listing/Dao.php (hunk 1 above)
Comment (BlackbitNeueMedien) [reply models/GridConfigShare/Listing/Dao.php]: Fixed.

File: bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Listing/Dao.php (hunk 3 above)
Comment (BlackbitNeueMedien) [reply bundles/EcommerceFrameworkBundle/CartManager/CartCheckoutData/Listing/Dao.php]: Fixed.

File: lib/Model/Listing/AbstractListing.php (hunk 5 above)
Comment (BlackbitNeueMedien) [reply lib/Model/Listing/AbstractListing.php:578]: Added return type. Anything more necessary? As this comes from the interface I would expect that people will lookup on https://www.php.net/manual/en/class.countable.php for more verbose description.

File: models/ImportConfigShare/Listing/Dao.php (hunk 6 above)
Comment (BlackbitNeueMedien) [reply models/ImportConfigShare/Listing/Dao.php]: Fixed.

File: bundles/EcommerceFrameworkBundle/CartManager/CartItem/Listing/Dao.php (hunk 4 above)
Comment (BlackbitNeueMedien) [reply bundles/EcommerceFrameworkBundle/CartManager/CartItem/Listing/Dao.php]: Fixed.

**Comment by dvesh3** [comment]
@BlackbitNeueMedien thanks a lot! 🙏

**Comment by dkarlovi** [comment]
I just came to report this issue as a feature request and it's already done! :rocket:

**Pull request merged**